consensus/service/src/validators.rs

```diff
@@ -37,64 +40,40 @@ impl<L: Ledger + Sync> DefaultTxManagerUntrustedInterfaces<L> {
 }
 
 impl<L: Ledger + Sync> TxManagerUntrustedInterfaces for DefaultTxManagerUntrustedInterfaces<L> {
-    /// Performs the untrusted part of the well-formed check.
-    /// Returns current block index and membership proofs to be used by
-    /// the in-enclave well-formed check on success.
+    /// Performs **only** the non-enclave part of the well-formed check.
+    ///
+    /// Returns the local ledger's block index and membership proofs for each highest index.
     fn well_formed_check(
         &self,
-        highest_indices: &[u64],
-        key_images: &[KeyImage],
-        output_public_keys: &[CompressedRistrettoPublic],
+        tx_context: &TxContext,
     ) -> TransactionValidationResult<(u64, Vec<TxOutMembershipProof>)> {
-        // The `key_images` must not have already been spent.
-        // Note that according to the definition at the top of this file, key image check is not part
-        // of the well-formedness check, but we do it anyway to more quickly get rid of transactions
-        // that are obviously unusable.
-        if key_images
```

Comment:
Checking if key images are in the ledger is an example of a check that could pass at one point in time, but could fail later (i.e. after more key images are added to the ledger). 

I think we previously had this here as an indirect way to give clients immediate feedback about whether their transaction would be rejected, but we now explicitly perform that check in the ClientApiService.